netket/experimental/operator/_particle_number_conserving_fermionic/_operators.py

```diff
@@ -0,0 +1,341 @@
+from functools import partial
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+
+from flax import struct
+
+from netket.operator import DiscreteJaxOperator
+from netket.hilbert import SpinOrbitalFermions
+from netket.utils.types import PyTree
+from netket.operator import FermionOperator2ndJax
+from netket.utils.optional_deps import import_optional_dependency
+
+from .._pyscf_utils import (
+    TV_from_pyscf_molecule,
+    to_desc_order_sparse,
+    compute_pyscf_integrals,
+)
+
+from .._normal_order_utils import to_normal_order, to_normal_order_sector
+from ._conversion import (
+    fermiop_terms_to_sites_daggers_weights,
+    fermiop_terms_to_sites_sectors_daggers_weights,
+    to_fermiop_helper,
+)
+from ._operator_data import (
+    collect_ops,
+    sparse_arrays_to_coords_data_dict,
+    prepare_operator_data_from_coords_data_dict,
+    prepare_operator_data_from_coords_data_dict_spin,
+    to_coords_data_sector,
+)
+from ._kernels import get_conn_padded_pnc, get_conn_padded_pnc_spin
+
+
+@struct.dataclass
```

Comment:
If tests pass, ok.
But as the base class is not a dataclass this might interplay badly, so if you see problem you might hve to implement by hand the to/from pytree methods...